crates/corro-agent/src/api/peer.rs

```diff
@@ -702,6 +711,53 @@ fn handle_need(
                 }
             }
         }
+        SyncNeedV1::Empty { ts } => {
+            if last_cleared_ts.is_none() {
+                return Ok(());
+            }
+            let last_cleared_ts = last_cleared_ts.unwrap();
+            let mut filter = String::from("");
+            if let Some(ts) = ts {
+                if ts >= last_cleared_ts {
+                    warn!("received empties timestamp {ts:?} greater our last cleared ts: {last_cleared_ts:?}");
+                    return Ok(());
+                }
+
+                filter = format!("ts > {ts}");
+            }
+
+            let query = format!("SELECT start_version, end_version, ts from __corro_bookkeeping WHERE actor_id = ? AND end_version IS NOT NULL ORDER BY ts {filter}");
```

Comment:
We might need an index on `__corro_bookkeeping.ts ASC` (maybe with `actor_id` too?). probably want to run a `EXPLAIN QUERY PLAN` on that to see what it uses.

Reply:
You can also use `crsql_site_id()` instead of manually passing the current node's actor_id